src/widgets/EmbedWidget.ts

```diff
@@ -78,62 +62,79 @@ export default class EmbedWidget extends Widget {
-        }
-        ctaElement.parentNode.removeChild(ctaElement);
-      }
 
       // @ts-ignore -- number will be cast to string by browsers
-      this.frame.height = frameDoc.body.scrollHeight;
+      frame.height = frameDoc.body.scrollHeight;
 
       // Adjust frame height when size of body changes
-      // @ts-ignore
+      /* istanbul ignore next: hard to test */
       const ro = new contentWindow["ResizeObserver"]((entries) => {
         for (const entry of entries) {
           const { height } = entry.contentRect;
           // @ts-ignore -- number will be cast to string by browsers
-          this.frame.height = height;
+          frame.height = height;
         }
       });
 
-      ro.observe(await this._findInnerContainer());
+      const container = await this._findInnerContainer(frame);
+      ro.observe(container);
 
-      // Regular load - trigger event
-      if (!this.context.container) {
+      if (this._shouldFireLoadEvent()) {
         this._loadEvent(_sqh);
         _log("loaded");
       }
     });
   }
 
-  // Un-hide if element is available and refresh data
+  /**
+   * Un-hide if element is available and refresh data
+   */
   open() {
-    if (!this.frame) return _log("no target element to open");
-    this.element.style.visibility = "unset";
-    this.element.style.height = "auto";
-    this.element.style["overflow-y"] = "auto";
+    const frame = this._findFrame();
+    if (!frame) return _log("no target element to open");
 
-    this.frame?.contentDocument?.dispatchEvent(new CustomEvent("sq:refresh"));
+    const element = this._findElement();
+
+    element.style.visibility = "unset";
+    element.style.height = "auto";
+    element.style["overflow-y"] = "auto";
+
+    frame?.contentDocument?.dispatchEvent(new CustomEvent("sq:refresh"));
     const _sqh =
-      this.frame?.contentWindow?.squatch ||
-      this.frame?.contentWindow?.widgetIdent;
-    this._loadEvent(_sqh);
-    _log("loaded");
+      frame?.contentWindow?.squatch || frame?.contentWindow?.widgetIdent;
+
+    if ((this.context as UpsertWidgetContext).user) {
+      this._loadEvent(_sqh);
+      _log("loaded");
+    }
   }
 
   close() {
-    if (!this.frame) return _log("no target element to close");
-    this.element.style.visibility = "hidden";
-    this.element.style.height = "0";
-    this.element.style["overflow-y"] = "hidden";
+    const frame = this._findFrame();
+    if (!frame) return _log("no target element to close");
+
+    const element = this._findElement();
+
+    element.style.visibility = "hidden";
+    element.style.height = "0";
+    element.style["overflow-y"] = "hidden";
     _log("Embed widget closed");
   }
 
   protected _error(rs, mode = "embed", style = "") {
     return super._error(rs, mode, style);
   }
+
+  private _shouldFireLoadEvent() {
+    const noContainer = !this.container;
+    const isComponent =
+      this.container instanceof HTMLElement &&
+      this.container.tagName.startsWith("SQUATCH-");
```

Comment:
```suggestion
      this.container.tagName.startsWith("SQUATCH-");
```
i think this also needs to handle `IMPACT-` if we support it

Reply:
Good catch